app/auth/signup/email/page.tsx

```diff
@@ -0,0 +1,113 @@
+'use client'
+import { useRef } from 'react'
+import { useForm } from 'react-hook-form'
+import {
+  Box,
+  Flex,
+  FormLabel,
+  FormErrorMessage,
+  FormControl,
+  Input,
+  Heading,
+  useToast
+} from '@chakra-ui/react'
+import { PrimaryButton } from '@/components/button'
+import { signUp } from './action'
+import { useFormResolver, SignUpSchema } from './schema'
+
+export default function SignUp() {
+  const toast = useToast()
+  const isLoading = useRef(false)
+  const toastId = useRef<ReturnType<typeof toast> | undefined>(undefined)
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<SignUpSchema>({
+    resolver: useFormResolver
+  })
+
+  return (
+    <>
+      <Heading>Welcome!</Heading>
+      <Heading>Create an Account</Heading>
+      <Box
+        as="form"
+        onSubmit={handleSubmit((data) => {
```

Comment:
I think it's better to make a function like `signUpHandler` not writing here to understand easily.　I personally think this is normal style if the function is big enough like this!